include/ensmallen_bits/problems/ackley_function_impl.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file ackley_function_impl.hpp
+ * @author Suryoday Basak
+ *
+ * Implementation of the Ackley function.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PROBLEMS_ACKLEY_FUNCTION_IMPL_HPP
+#define ENSMALLEN_PROBLEMS_ACKLEY_FUNCTION_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "ackley_function.hpp"
+
+namespace ens {
+namespace test {
+
+inline AckleyFunction::AckleyFunction(const double epsilon) : epsilon(epsilon)
+{ /* Nothing to do here */ }
+
+inline void AckleyFunction::Shuffle() { /* Nothing to do here */ }
+
+inline double AckleyFunction::Evaluate(const arma::mat& coordinates,
+                                       const size_t /* begin */,
+                                       const size_t /* batchSize */) const
+{
+  // For convenience; we assume these temporaries will be optimized out.
+  const double x1 = coordinates(0);
+  const double x2 = coordinates(1);
+
+  const double objective = -20 * std::exp(-0.2 * std::sqrt(0.5 * (std::pow(x1, 2) + 
+				 std::pow(x2, 2)))) - 
+	  			 std::exp(0.5 * (std::cos(2 * arma::datum::pi * x1) + 
+				 std::cos(2 * arma::datum::pi * x2))) +
+				 std::exp(1) + 20;
+
+  return objective;
+}
+
+inline double AckleyFunction::Evaluate(const arma::mat& coordinates) const
+{
+  return Evaluate(coordinates, 0, NumFunctions());
+}
+
+inline void AckleyFunction::Gradient(const arma::mat& coordinates,
+                                     const size_t /* begin */,
+                                     arma::mat& gradient,
+                                     const size_t /* batchSize */) const
+{
+  // For convenience; we assume these temporaries will be optimized out.
+  const double x1 = coordinates(0);
+  const double x2 = coordinates(1);
+
+  //Aliases for different terms in the expression of the gradient
+  const double t0 = std::sqrt(0.5 * (std::pow(x1, 2) + std::pow(x2, 2)));
+  const double t1 = 2.0 * std::exp(-0.2 * t0)/(t0 + epsilon);
+  const double t2 = arma::datum::pi*std::exp(0.5 * (std::cos(2 * arma::datum::pi * x1)
```

Comment:
Picky style comment, can you use a space between operations? And use 4 spaces for the next line? See https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping for more details.

Reply:
I'm sorry we keep running into this, I'll fix this as required.